3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```diff
@@ -1,27 +1,21 @@
 'use strict';
 const body = document.querySelector('body');
 async function requestData(url) {
-  try {
-    const response = await fetch(url);
-    if (response.ok) {
-      const data = await response.json();
-      return data;
-    } else {
-      throw new Error();
-    }
-  } catch (error) {
-    throw error;
+  const response = await fetch(url);
+  if (response.ok) {
+    const data = await response.json();
+    return data;
+  } else {
+    throw new Error(`HTTP Error ${response.status}: ${response.statusText}`);
   }
```

Comment:
There is no need for an `else` block here because if the `if` block executes the function will return early without the `throw` statement ever coming into play.

```suggestion
  throw new Error(`HTTP Error ${response.status}: ${response.statusText}`);
```